File: tournaments/replaceMiddle/replaceMiddle.js
function replaceMiddle(arr) {
    const n = arr.length;

    if (n & 1) return arr;

    const halfLength = ~~(n / 2) - 1;

    arr[halfLength] = arr[halfLength] + arr[halfLength + 1];
    arr.splice(halfLength + 1, 1);

    return arr;
}

File: tournaments/replaceMiddle/replaceMiddle.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub replaceMiddle {
    my ($arr) = @_;

    my $len = scalar(@$arr);
    my $halfLen = int($len / 2);

    unless ($len & 1) {
        @$arr[$halfLen] = $$arr[$halfLen - 1] + $$arr[$halfLen];
        splice($arr, $halfLen - 1, 1);
    }

    return $arr;
}
